design-proposals/eim-platform-bundle-improvements.md

```diff
@@ -0,0 +1,133 @@
+
+Author(s): Tomasz Osiński
+
+Last updated: 14.05.2025
+
+## Abstract
+
+Platform Bundle is defined as a collection of OS-level files, installation scripts, configurations and packages.
+It's a concept to make any OS profile supported by EIM extensible and customizable. For example,
+we can have different flavors of Ubuntu/EMT images that support various use cases.
+It's a mechanism that allows EN support features beyond what EIM supports currently.
+
+Platform Bundle is already integrated with EIM and gives ability to provide custom cloud-init template
+and Installer script (for Ubuntu).
+For instance, the `Ubuntu with Intel extensions` OS profile uses Installer script that is curated
+by EEF (Edge Enablement Framework) to install additional GPU drivers.
+
+However, the current Platform Bundle design has the following problems:
+
+1. Co-ownership of cloud-init template and Installer script between EIM and EEF,
+   leading to de-synchronization between the two and error-prone manual processes to update
+   Platform Bundle scripts any time cloud-init or Installer script is changed in the EIM codebase.
+2. Currently, Platform Bundle is limited to cloud-init template and Installer script,
+   while the original concept of Platform Bundle was to make it any OS-level extension.
+
+This design proposal aims at:
+
+- Define clear ownership of cloud-init template and Installer script between EIM and EEF,
+  without impacting EEF certification process.
+- Extend the Platform Bundle integration to enable providing custom OS-level files and configurations.
+
+## Proposal
+
+The proposal is inspired by two approaches from existing solutions:
+
+- Azure and VM template with `customize` section
+- VMware Tanzu BYOH with `OS bundle` concept
+
+This design makes the following proposals:
+
+**EIM:**
+
+- Sets the sole ownership of EIM cloud init template to EIM codebase.
+  It is NOT possible to overwrite EIM cloud-init template via the Platform Bundle,
+  because EIM cloud-init is foundational piece that, if mis-configured, may make the entire EMF unfunctional.
+- It is still possible to overwrite EIM Installer script for Ubuntu via the Platform Bundle.
+  This may be needed if we want support other OS distros in the future.
+- Extracts EIM cloud-init template to a separate sub-module of infra-onboarding.
+  The new sub-module is versioned separately following SemVer.
+  EIM cloud-init template is published to Release Service as version OCI artifact.
+- OM imports EIM cloud-init template as versioned Go module.
+- EIM extends the implementation of Platform Bundle to supply provisioned OS
+  with arbitrary files defined as part of the Platform Bundle.
+  It implies change of the platform bundle YAML format.
+- `platformBundle` field of the OS resource should become R/W field from northbound API,
+  so that external users can create their own custom OS profiles with arbitrary files packages.
```

Comment:
For these custom OS profiles that users create, is it expected that the files listed in the platformBundles field have already been loaded into the release server by the user or will OM need to pull from the location listed and push them to the release service for future use? If it is the first option that will be used, are there changes needed for the release service to support this as well?